### ISSUE TITLE
Local changes not used when building

### ISSUE THREAD

**Issue by verdie-g** [dotnet/runtime#42526]
Hello,

I'm trying to build the runtime to be able to benchmark my PR (#42377) with
```
git clone https://github.com/dotnet/runtime
cd runtime
gh pr checkout 42377
buid.cmd -rc Release -c Release
```
After a successful build, I used DotPeek to check that `artifacts\bin\testhost\net5.0-Windows_NT-Release-x64\shared\Microsoft.NETCore.App\6.0.0\System.Threading.Tasks.dll` was correctly generated but the `Task.WhenAll` method still has the old code without my changes.

I have tried following the [building libraries guide](https://github.com/dotnet/runtime/tree/master/docs/workflow/building/libraries) with no success either.

I suppose that I've missed something but I cannot figure out what, so I'd like to discuss how to make the documentation clearer.

**Comment by stephentoub** [comment]
> System.Threading.Tasks.dll

Task.WhenAll is implemented in System.Private.CoreLib.  What assembly are you looking at?

**Comment by ericstj** [comment]
Agree with @stephentoub, your PR https://github.com/dotnet/runtime/pull/42377 modifies System.Private.CoreLib.dll, not System.Threading.Tasks.dll.  System.Threading.Tasks.dll doesn't contain any `Task` type.

https://github.com/dotnet/performance/blob/master/docs/benchmarking-workflow-dotnet-runtime.md has steps we typically use for running benchmarks against dotnet/runtime.  cc @adamsitnik

**Comment by verdie-g** [comment]
> Task.WhenAll is implemented in System.Private.CoreLib

![image](https://user-images.githubusercontent.com/9092290/93791859-0bd20180-fc35-11ea-8795-4c9f84558d5c.png)
Apparently, `System.Threading.Tasks` references `System.Private.CoreLib` and when I opened the reference I couldn't find my code. However, I've just opened directly `System.Private.CoreLib` and my work is present. Not sure to understand the difference but thanks :)

The reason I was wondering if the assembly was correctly generated with my changes is actually because an error in my benchmark.
